core/authority-discovery/Cargo.toml

```diff
@@ -0,0 +1,32 @@
+[package]
+name = "substrate-authority-discovery"
+version = "2.0.0"
+authors = ["Parity Technologies <admin@parity.io>"]
+edition = "2018"
+build = "build.rs"
+
+[build-dependencies]
+prost-build = "0.5"
+
+[dependencies]
+network = { package = "substrate-network", path = "../../core/network" }
+sr-primitives = { path = "../../core/sr-primitives" }
+primitives = { package = "substrate-primitives", path = "../primitives" }
+client = { package = "substrate-client", path = "../../core/client" }
+authority-discovery-primitives = { package = "substrate-authority-discovery-primitives", path = "./primitives", default-features = false }
+codec = { package = "parity-scale-codec", default-features = false, version = "1.0.3" }
+futures = "0.1.17"
```

Comment:
No particular reason to using `0.1.17` other than other packages using the same:

```bash
grep -ri --include Cargo.toml "futures = "
node-template/Cargo.toml:futures = "0.1"
core/client/Cargo.toml:futures = { version = "0.1", optional = true }
core/network/Cargo.toml:futures = "0.1.17"
core/authority-discovery/Cargo.toml:futures = "0.1.17"
core/service/test/Cargo.toml:futures = "0.1"
core/service/Cargo.toml:futures = "0.1.17"
core/finality-grandpa/Cargo.toml:futures = "0.1"
core/cli/Cargo.toml:futures = "0.1.17"
core/rpc/Cargo.toml:futures = "0.1.17"
core/consensus/rhd/Cargo.toml:futures = "0.1.17"
node/rpc-client/Cargo.toml:futures = "0.1.26"
node/cli/Cargo.toml:futures = "0.1"
Cargo.toml:futures = "0.1"
```

Shouldn't we update them all at once?

More general question: Why are we ever specifying the semver bug fix version?

---

> As an aside, I think it would be great if dependencies were sorted alphabetically.

I will do that.

Reply:
> More general question: Why are we ever specifying the semver patch version?

Considering that this is below 1.0, the general rule is that the patch version here is the equivalent of a minor version post-1.0.

If you use a feature that was added in `futures` 0.1.12 and you only depend on `futures` 0.1.11 for example, your code might compile thanks to your Cargo.lock but you still have things wrong.